Jenkinsfile

```diff
@@ -899,6 +899,8 @@ pipeline {
                     }
                     post {
                         always {
+                            discoverGitReferenceBuild referenceJob: 'daos-stack/daos/master',
```

Comment:
Yes, that's happening today because of https://github.com/daos-stack/pipeline-lib/blob/master/vars/unitTestPost.groovy#L90-L91

You only need to call discoverGitReferenceBuild and normally it gets called as part of NLT, but if you skip that stage then fault injection does not have a value set so it uses the previous build of the PR as a reference which doesn't work either.  This does need fixing properly, however I've not attempted that in this PR.